yt_dlp/extractor/polskieradio.py

```diff
@@ -322,22 +353,74 @@ def _entries(self, url, page, category_id):
                 yield self.url_result(
                     compat_urlparse.urljoin(url, href), PolskieRadioLegacyIE,
                     entry_id, entry.get('title'))
-            mobj = re.search(
-                r'<div[^>]+class=["\']next["\'][^>]*>\s*<a[^>]+href=(["\'])(?P<url>(?:(?!\1).)+)\1',
-                content)
-            if not mobj:
-                break
-            next_url = compat_urlparse.urljoin(url, mobj.group('url'))
-            content = self._download_webpage(
-                next_url, category_id, 'Downloading page %s' % page_num)
+            for a_entry in re.findall(r'<span data-media=({[^ ]+})', content):
+                media = self._parse_json(a_entry, category_id)
+                yield {
+                    'url': media['file'],
+                    'id': media.get('uid'),
+                    'title': unquote(media.get('title')),
+                    'description': unquote(media.get('desc')),
+                    'duration': media.get('length'),
+                }
+            if is_billennium_tabs:
+                mobj = re.search(
+                    r'<div[^>]+class=["\']next["\'][^>]*>\s*<a[^>]+onclick=(["\'])TB_LoadTab\((?P<params>(?:(?!\1).)+)\);\1',
+                    pagination)
+                if not mobj:
+                    break
+                params = self._parse_json('[' + js_to_json(unescapeHTML(mobj.group('params'))) + ']', category_id)
+                tab_content = self._download_json(
+                    'https://www.polskieradio.pl/CMS/TemplateBoxesManagement/TemplateBoxTabContent.aspx/GetTabContent',
+                    category_id, 'Downloading page %s' % page_num, data=json.dumps({
+                        'tabId': params[1],
+                        'boxInstanceId': params[0],
+                        'sectionId': params[3],
+                        'categoryId': params[4],
+                        'categoryType': params[2],
+                        'subjectIds': params[6],
+                        'tagIndexId': params[7],
+                        'queryString': params[8],
+                        'name': params[9],
+                        'pageNumber': page_num,
+                        'pagerMode': params[5],
+                        'openArticlesInParentTemplate': params[10],
+                        'idSectionFromUrl': params[11],
+                        'maxDocumentAge': params[12],
+                        'showCategoryForArticle': params[13],
+                    }).encode('utf-8'), headers={'content-type': 'application/json'})
```

Comment:
```suggestion
                tab_content = self._download_json(
                    'https://www.polskieradio.pl/CMS/TemplateBoxesManagement/TemplateBoxTabContent.aspx/GetTabContent',
                    category_id, f'Downloading page {page_num}',
                    headers={'content-type': 'application/json'}, data=json.dumps({
                        'pageNumber': page_num,
                        **dict(zip((
                            'boxInstanceId', 'tabId', 'categoryType', 'sectionId', 'categoryId', 'pagerMode',
                            'subjectIds', 'tagIndexId', 'queryString', 'name', 'openArticlesInParentTemplate',
                            'idSectionFromUrl', 'maxDocumentAge', 'showCategoryForArticle',
                        ), params),
                    }).encode())
```
would be way cleaner imo